### PR TITLE
SW-3839 Plants Dashboard Map Tooltips

### PR DESCRIPTION
**Pull request by charlie-tf**
- cleaned up some reusable logic into selectors
- add tooltips
- fix mortality rate transparency logic

image
image
image

### CODE REVIEW

File: src/redux/features/tracking/trackingSelectors.ts [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
 import { PlantingSite } from 'src/types/Tracking';
+import { createCachedSelector } from 're-reselect';
 
 export const selectPlantingSites = (state: RootState) => state.tracking?.plantingSites;
 export const selectPlantingSitesError = (state: RootState) => state.tracking?.error;
 
 export const selectPlantingSite = (state: RootState, plantingSiteId: number) =>
   selectPlantingSites(state)?.find((site: PlantingSite) => site.id === plantingSiteId);
+
+export const selectZoneProgress = createCachedSelector(
```
Comment (karthikbtf) [inline src/redux/features/tracking/trackingSelectors.ts:11]: @charlie-tf just curious.. can this be used in @constanzauanini 's PR later?
See the BE [PR](https://github.com/terraware/terraware-server/pull/1293) that was updated for the nursery withdrawals plantings redux WIP [PR](https://github.com/terraware/terraware-web/pull/1558).

File: src/redux/features/tracking/sitePopulationSelector.ts [hunk 2]
```diff
@@ -1,5 +1,26 @@
...
+          const count = +population['totalPlants(raw)'];
+          numPlants += isNaN(count) ? 0 : count;
+          speciesSet.add(population.species_scientificName);
+        })
+      );
+      zoneStats[+zone.id] = { name: zone.name, reportedPlants: numPlants, reportedSpecies: speciesSet.size };
+    });
+    return zoneStats;
+  }
+)(() => 'zonePopulationStats');
```
Comment (karthikbtf) [inline src/redux/features/tracking/sitePopulationSelector.ts:26]: for this use-case we can use `createSelector` which maintains a cache of 1 entry.
CachedSelector was introduced to support multiple cache entries (helpful when the cache key is dynamic).

**Review by karthikbtf — APPROVED** [review]
Looks swell!! One suggestion and one question.
🎉

File: src/redux/features/tracking/trackingSelectors.ts (hunk 1 above)
Comment (constanzauanini) [reply src/redux/features/tracking/trackingSelectors.ts:11]: I think this is by zone, I need it by subzone

Comment (charlie-tf) [reply src/redux/features/tracking/trackingSelectors.ts:11]: Yeah, this selector aggregates by zone

**Review by karthikbtf — APPROVED** [review]
thanks for the change, looks good 👍

**Pull request merged**